tests/test_ops.py

```diff
@@ -256,3 +225,24 @@ def test_create_ops_arg_function(self, read):
                 Literal('"%s"' % dtype_to_cstr(u.dtype)),
                 namespace['ops_read'] if read else namespace['ops_write']
             ]
+
+    @pytest.mark.parametrize('equation,expected', [
+        ('Eq(u.forward, u + 1)',
+         '[\'ops_dat_fetch_data(u_dat[(time_M)%(2)],0,&(u[(time_M)%(2)]));\','
+         '\'ops_dat_fetch_data(u_dat[(time_M + 1)%(2)],0,&(u[(time_M + 1)%(2)]));\']'),
+        ('Eq(v, v.dx + u)',
+         '[\'ops_dat_fetch_data(v_dat[(time_M)%(2)],0,&(v[(time_M)%(2)]));\','
+         '\'ops_dat_fetch_data(v_dat[(time_M + 1)%(2)],0,&(v[(time_M + 1)%(2)]));\','
+         '\'ops_dat_fetch_data(u_dat[(time_M)%(2)],0,&(u[(time_M)%(2)]));\','
+         '\'ops_dat_fetch_data(u_dat[(time_M + 1)%(2)],0,&(u[(time_M + 1)%(2)]));\']'),
+    ])
+    def test_create_fetch_data(self, equation, expected):
+
+        grid = Grid(shape=(4, 4))
+
+        u = TimeFunction(name='u', grid=grid)  # noqa
```

Comment:
May I suggest to try also with:

- more time_orders, ie `TimeFunction(..., time_order=x)`
- equations using time derivatives such as iso acoustic (`Eq(u.forward, u.dt ....)`)

Reply:
Sure! I'll put some new tests.